### PR TITLE
test/autodjprocessor_test: Fix DEBUG_ASSERT caused by duplicated COs

### PR DESCRIPTION
**Pull request by Holzhaus**

### CODE REVIEW

**Review by uklotzde — CHANGES_REQUESTED** [review]
Unfortunately their seem to be much more issues regarding the creation and management of ControlObject instances. After a quick peek at the code I already found a bug.

Another aspect: Running tests that contain control objects must not be run concurrently. Though I think this is the case.

**Comment by Holzhaus** [comment]
> Another aspect: Running tests that contain control objects must not be run concurrently. Though I think this is the case.

Does mixxx-test run tests concurrently? I thought not. And ctest does run tests in parallel, but it spawns a new separate process for each test, so this should not be an issue.

File: src/test/autodjprocessor_test.cpp [hunk 1]
```diff
@@ -202,6 +202,11 @@ class AutoDJProcessorTest : public LibraryTest {
         EXPECT_CALL(*pPlayerManager, getPlayer(QString("[Channel3]"))).Times(1);
         EXPECT_CALL(*pPlayerManager, getPlayer(QString("[Channel4]"))).Times(1);
 
+        // We need to call reset *before* constructing a new
+        // MockAutoDJProcessor, because otherwise the new object will try to
+        // create COs that already exist because they were created by the
+        // previous instance.
+        pProcessor.reset();
```
[the remark this answers is not in the document]
Comment (Holzhaus) [reply src/test/autodjprocessor_test.cpp:209]: Whoops, you're right. I searched for other occurrences and this looked like the code below. I'll remove it.

**Comment by uklotzde** [comment]
> > Another aspect: Running tests that contain control objects must not be run concurrently. Though I think this is the case.
> 
> Does mixxx-test run tests concurrently? I thought not. And ctest does run tests in parallel, but it spawns a new separate process for each test, so this should not be an issue.

No. The word "not" got lost in my last sentence.

**Comment by uklotzde** [comment]
LGTM

**Pull request merged**